vncviewer/DesktopWindow.cxx

```diff
@@ -322,6 +324,29 @@ void DesktopWindow::setCursor(int width, int height,
 }
 
 
+void DesktopWindow::setCursorPos(const rfb::Point& pos)
+{
+  if (!mouseGrabbed) {
+    // Do nothing if we do not have the mouse captured.
+    return;
+  }
+#if defined(WIN32)
+  SetCursorPos(pos.x + x_root() + viewport->x(),
+               pos.y + y_root() + viewport->y());
+#elif defined(__APPLE__)
+  CGPoint new_pos;
+  new_pos.x = pos.x + x_root() + viewport->x();
+  new_pos.y = pos.y + y_root() + viewport->y();
+  CGWarpMouseCursorPosition(new_pos);
+#else // Assume this is Xlib
+  Window rootwindow = DefaultRootWindow(fl_display);
+  XWarpPointer(fl_display, rootwindow, rootwindow, 0, 0, 0, 0,
+               pos.x + x_root() + viewport->x(),
+               pos.y + y_root() + viewport->y());
```

Comment:
Have you tried this with Wayland? It doesn't seem to work for me here. :/

Reply:
no, and i don't have a wayland environment :(